### PR TITLE
Doc link fixes & enhancements to Bearer token authentication tutorial

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR fixes links, some grammar, and section numbering as per the preferred Quarkus Contributor guidelines for writing tutorials. 
This also makes the content more harmonious with product docs downstream for the 3.2 release.

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@gastaldi / @gsmet - I'm not seeing the doc site preview yet in this PR. The sad and empty wine glass emoji doesn't even display :wine_glass: Is it still poorly?  Thanks

**Comment by michelle-purcell** [comment]
@sberyozkin - Please can you review the updates here :pray: 

I made some style and grammar enhancements after running the product doc prep style checker and final PV2 preview check. I hope these don't alter technical meaning but add clarity, some accessibility, and localization benefits.
I remember making these before but I think they got wiped out somehow during conflict resolution :confused: 

I also added the (harmonious) section numbering approach and patch for our PV2 system, which was recently tried and tested with the basic auth tutorial.

If you spot anything here that isn't applicable to 3.2, please let me know. Thanks :-)

**Comment by gastaldi** [comment]
@michelle-purcell that's because our CI is still busy, you should hopefully see it when the job completes 😀

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:316]:
```suggestion
For information about testing your application in devmode, see the preceding <<keycloak-dev-mode>> section.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:316]: I think `devmode` is a real word in the Quarkus vocabulary, @gastaldi @geoand is it the case ? I'm not sure `developer mode` works

**Comment by sberyozkin** [comment]
Thanks @michelle-purcell, looks nice, I only have a question about `devmode` vs `developer mode`, my understanding `devmode` is preferred but lets wait for some comments, thanks

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (geoand) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:316]: I see us using `development mode` in other docs so it's probably best to be consistent

Comment (gastaldi) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:316]: +1 to `Development mode`

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:35]:
```suggestion
Because it is an OAuth 2.0 Authorization server, the token also references the client acting on the user's behalf.
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:34]:
```suggestion
A Keycloak server issues the bearer token and represents the subject for which the token was issued.
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:40]:
```suggestion
The `/api/admin` endpoint is protected with RBAC (Role-Based Access Control), which only users with the `admin` role can access.
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:41]:
```suggestion
At this endpoint, the `@RolesAllowed` annotation is used to enforce the access constraint declaratively.
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:48]:
```suggestion
You can clone the Git repository by running the command `git clone {quickstarts-clone-url}`, or you can download an {quickstarts-archive-url}[archive].
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:180]:
```suggestion
The main difference in this example is that the `@RolesAllowed` annotation is used to verify that only users granted the `admin` role can access the endpoint.
```

**Review by MichalMaler — APPROVED** [review]
Hey, @michelle-purcell Here are just some minor suggestions. Apply if you like them. The rest is good. Approving! :) 
Cheers!

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:180]: or the `_users with the `admin` role can access._` to be consistent with my first suggestion.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc]: @sberyozkin , @gastaldi, @geoand - Thanks for flagging this and your comments.

As 'dev mode' is a proper noun defined by Quarkus and commonly used in this specialization,  I have now reverted back to `dev mode` for all instances except for the first one. 

My :tropical_fish: brain forgot about a similar discussion with Quarkus SMEs that came up earlier this year:

We concluded that to prevent ambiguity around the word `dev` (is it developer/development, it would be best practice to use the full form in the first instance on the page, for example, `development (dev) mode`, and use `dev mode` thereafter.

I will also clean up of other inconsistent instances across the other docs later **after 3.2 product GA** so that we don't create backport headaches. 

I updated the PR according to ^^. Please do shout if you disagree with this approach. Thanks again :green_heart:

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:34]: @MichalMaler - This suggestion changes the technical meaning. 

If the _Keycloak server represents the subject for which the token was issued_ and not the _bearer token_, then I will commit this change.

@sberyozkin - WDYT?

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:180]:
```suggestion
The main difference in this example is that the `@RolesAllowed` annotation is used to verify that only users with the `admin` role can access the endpoint.
```

**Comment by michelle-purcell** [comment]
@gastaldi / @gsmet - Any ideas why the doc build is failing in this and other PRs today?

 I'm seeing several errors like these in my PR and some others too.

image
Thanks

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc]: Thanks @michelle-purcell @geoand @gastaldi

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc:34]: @michelle-purcell Good catch, yes, Michal's suggestion changes the technical meaning

**Review by sberyozkin — APPROVED** [review]

**Review by geoand — APPROVED** [review]

**Comment by michelle-purcell** [comment]
@gastaldi / @sberyozkin - As we have approval from 3 folks, could we merge this PR before it becomes stale and in conflict with other PRs?

**Comment by gastaldi** [comment]
@michelle-purcell can you please squash all commits before we merge it?

**Comment by michelle-purcell** [comment]
@gastaldi - Sorry forgot to do that! It's done now. Thanks :+1:

**Comment by michelle-purcell** [comment]
@gastaldi :superhero:  Thank you.

**Pull request merged**
